[Feature]: Added level progress bar

powerpuff-luv opened the pull request:
<!--
🎉 Thanks for contributing to Dinosaur Exploder!

🚩⚠️ Before submitting your pull request, look at the different hidden HTML comments "<!-- ..." to guide how on to write this Pull Request and understand how to respect the guidelines of the Pull Request template.

After you open the PR, a maintainer will need to approve the GitHub Actions before they can run.
Once approved, GitHub will automatically check if the project builds and runs correctly.
You’ll also see a message from the GitHub bot with a downloadable .jar file — so you can test the game directly from your own work! Pretty cool, right? 😊
-->

### ✅ PR Checklist

> [!TIP]
> Please check the boxes below to confirm you followed the contribution guidelines:

<!-- ✅ To check a box, write: `[x]` (no space between the brackets) -->

- [x] My PR title follows the format: `[Type]: clear description of change`
- [x] I used the correct `type`: Feature, Fix, Refactor, Docs, Test, Chore
- [x] I reviewed my code and removed unnecessary debug logs or comments.
- [x] I tested my changes and verified they work as expected.
- [x] I ran the project to confirm it compiles and runs correctly.
- [x] I read the [Code of Conduct](https://github.com/jvondermarck/dinosaur-exploder/blob/main/CODE_OF_CONDUCT.md) and the [Contribution Guidelines](https://github.com/jvondermarck/dinosaur-exploder/blob/main/CONTRIBUTING.md).


### 📝 What does this PR do?

<!-- Briefly describe what you've changed or added -->
<!-- Example: Added a new dinosaur class with special abilities and updated the main menu to include it. -->

- This PR adds a Level Progress Bar to the game, which shows the progress of regular levels, and hides during boss levels.
- The Progress Bar requires a Level Manager to be created first, so it is important that now in the DinosaurController class in the initGame() method the levelManager is created first, and only then all Game Entities.

### 🔗 Related Issue

<!-- To close an issue automatically when the PR is merged, use the format: `Fixes #<issue-number>` -->
<!-- ✅ To check a box, write: `[x]` (no space between the brackets) -->

- [x] This PR fixes/closes: #168
- [ ] This PR doesn’t address a specific issue.

### 📸 Screenshots / Demos (if applicable)


https://github.com/user-attachments/assets/7b9300c5-9735-45c3-80c6-3863eb10265c



### 💬 Extra Notes (Optional)

<!-- Anything else you'd like to add? Questions, a review about what you think of this repo, what it could be improved, how did you find this repo... Any fun facts you want -->

<!-- Please delete the two lines below (used by default) if you have something to say. And write your message. -->
> [!NOTE]
> 🦕 No extra notes.

---

❤️ Thanks again for your contribution to Dinosaur Exploder!  
Your efforts help make this project awesome for everyone. 🦖✨

powerpuff-luv committed:
Added level progress bar. Now LevelManager creates before Game Entities.

powerpuff-luv committed:
Merge main into feature/level-progress-bar

jvondermarck reviewed (changes requested):
I really like it, great idea BUT on your video you can see a bug, when you go to level 2, the text takes some seconds to update from "Level 1" to "Level 2" so it is very weird. Can you fix this ? Otherwise great work.

You can see it on your screen recording at 00:05 seconds

powerpuff-luv committed:
Resolved bug with resetting level progress bar without delay. Added animation for level progress bar updates.

powerpuff-luv commented:
https://github.com/user-attachments/assets/7bb7a284-d203-4b3a-9bf8-2ca07ec6e501

I fixed a bug, now the progress bar remains filled until the next level starts.
I also added animations for the progress bar updates.

But there is something important. 
During the two-second pause between levels, LevelManager records all hits made by player. And by the time the next level starts, the progress of that level may not be zero, as shown at the end of the video.

It even looks like it was meant to be that way. In any case, it looks like the progress bar work is finished.

jvondermarck reviewed (approved):
Looks better now, thank you so much !
